src/services/productService.ts

```diff
@@ -0,0 +1,105 @@
+// Import product model
+import { Product } from "../models/ProductModel";
+
+// Import rating model
+import { Rating } from "../models/RatingModel";
+
+// Import Sequelize and Op from sequelize module
+import { Sequelize, Op } from "sequelize";
+
+export class productService {
+  // This method to add discount information to product information
+  static addDiscountInfo(productInfo: any) {
+    if (productInfo.discount_percentage === 0) {
+      productInfo.price_after_discount = null;
+      return;
+    }
+    const discountValue =
+      productInfo.price * (productInfo.discount_percentage / 100);
```

Comment:
constant numbers should be moved to an external file - that would be more proper in order for changes to take effect upon the whole project